rfcs/036-ml-feature-discovery.md

```diff
@@ -0,0 +1,195 @@
+- Feature Name: ml_feature_discovery
+- Start Date: 2021-05-05
+- RFC PR: [amundsen-io/rfcs#36](https://github.com/amundsen-io/rfcs/pull/36) (after opening the RFC PR, update this with a link to it and update the file name)
+- Amundsen Issue: [amundsen-io/amundsen#0000](https://github.com/amundsen-io/amundsen/issues/0000) (leave this empty for now)
+
+# ML Features Discovery in Amundsen
+
+## Summary
+
+Amundsen currently indexes a few first-class entities, however we currently do not index any ML features that are easily discoverable for users. The goal of this work is to index ML features at the same level as Tables, Dashboards, and People resources in Amundsen.
+
+## Motivation
+
+Integration of ML projects in Amundsen has been heavily requested by the community and has already been implemented by other similar projects. This RFC outlines what will be the first step towards indexing ML projects in Amundsen since feature engineering is one of the most time consuming processes for ML engineers. Providing them with the ability to easily discover features that they can reuse for their models will significatly increase their productivity.
+
+## Guide-level Explanation (aka Product Details)
+
+### New Metadata Definitions
+#### Semantic Metadata
+Metadata related to the feature itself.
+- Key: what makes a feature uniquely identifiable is the group, name and version. We don’t use db/source because the same feature can exists in different places (like hive and dynamo)
+    - ex: test_feature_group_name/test_feature_name/1.2.0
+- Availability/sources: list of places where feature is available
+- Name: the name of the feature
+- Version: the version of the feature
+- Data type
+- Description: optional, a description of the feature
+- Owners: individual and team emails of owners
+- Feature group: a way of grouping features, can be defined differently in each company, in this case we assume this will be the name of the pipeline that produced the features
+- Entity: a way of grouping features semantically by using nouns
+- Owner tags: programmatically generated and not editable tags, useful to group features base don the specific engineering team using them
+- User tags: normal amundsen tags
+- Contains PII: weather a feature contains PII or not, will be a badge in the badges list
+- Date range: what time frame is the data in the feature for
+- Partition column: optional column associated to the feature to indicate partition
```

Comment:
badge?

Reply:
if you read below you'll see that we are leveraging badge on the model to indicate this

Reply:
Yes PII will be a badge!